Various enhancements related to menu naming, READMEs...

echarles opened the pull request:
This PR brings various enhancements related to menu naming, READMEs...

No feature is added/removed, but we try to make it easier for the user to find its way in the UI when he runs multiple examples at the same time.

jtpio commented on .gitignore:
```diff
@@ -6,3 +6,4 @@ node_modules/
 .ipynb_checkpoints
 *.tsbuildinfo
 *.lock
+**/node_modules/**/README.md
```
Looks like this should be two separate lines.

echarles replied:
embedme was picking node_modules/rimraf/README.md and printing warning, asking to add in .gitignore if we wanted to skip that. I will add a comment to this line to make it more clear.

jtpio replied:
Ah right indeed, then we should keep it like this :+1:

jtpio commented on README.md:
```diff
@@ -57,80 +53,110 @@ You can expect from each example:
 
 ### [Main Menu](main-menu)
 
+Add a Menu in to the main app.
```
```suggestion
Add a Menu to the main app.
```

jtpio commented on README.md:
````diff
@@ -184,15 +210,37 @@ jlpm run build
 ```
 
-Start JupyterLab in watch mode:
+You can now start JupyterLab and check if your extension is work fine:
````
```suggestion
You can now start JupyterLab and check if your extension is working fine:
```

jtpio commented on README.md:
```diff
@@ -184,15 +210,37 @@ jlpm run build
+From there, you can change your extension source code, it will be recompiled,
+and you can refresh your browser to see your changes.
+
+We are using [embedme](https://github.com/zakhenry/embedme) to embed code snippets into the markdown READMEs. If you make changes to the source code, ensure you update the README and run `yarn embedme` from the root of the repository to regenerate the READMEs.
```
Should we mention `jlpm` instead of `yarn`?

echarles replied:
Done. Thx

jtpio commented on advanced/kernel-output/README.md:
```diff
@@ -227,8 +229,8 @@ and display:
 commands.addCommand(CommandIDs.execute, {
-  label: 'kernel-output: Execute Code',
-  caption: 'Execute Code',
+  label: 'Contact Kernel and Execute Code',
```
How about connect instead of contact?

```suggestion
  label: 'Connect to the Kernel and Execute Code',
```

jtpio commented on basics/datagrid/README.md:
```diff
@@ -1,9 +1,13 @@
 
-Jupyterlab is built on top of
-[Lumino](https://github.com/jupyterlab/lumino). That library defines `Widget` as the primary interface brick. In this example [the datagrid lumino example](https://jupyterlab.github.io/lumino/datagrid/index.html)
+Jupyterlab is built on top of [Lumino](https://github.com/jupyterlab/lumino).
```
```suggestion
JupyterLab is built on top of [Lumino](https://github.com/jupyterlab/lumino).
```

jtpio commented on basics/hello-world/README.md:
```diff
@@ -141,8 +142,9 @@ jupyter labextension link .
-second step runs the build script. In this step, the typescript code gets
+components that you want to use in your project.
+
+The second step runs the build script. In this step, the typescript code gets
```
```suggestion
The second step runs the build script. In this step, the TypeScript code gets
```

jtpio commented on basics/signals/README.md:
```diff
@@ -9,74 +11,78 @@
 ([documentation](https://jupyterlab.github.io/lumino/api/signaling/globals.html)).
 
-The basic concept is the following: a widget, in this case the one that contains
+The basic concept is the following.
```
Would this be more natural?

```suggestion
The basic concept is as follows:
```

jtpio commented on basics/signals/README.md:
````diff
@@ -9,74 +11,78 @@
 ```
 
-Another widget, in this case the panel that boxes several different widgets,
+Another widget, in this case the panel (`panel.ts`)that boxes several different widgets,
````
```suggestion
Another widget, in this case the panel (`panel.ts`) that boxes several different widgets,
```

jtpio commented on basics/signals/README.md:
```diff
@@ -94,41 +100,44 @@ export class ExampleView extends ReactWidget {
-a big red button is clicked. It is not very spectacular but the signaling is
-conceptually important for building extensions.
+The final extension writes a little `Button is clicked.` text to the browser console and on an alter when
+a big red button is clicked.
```
```suggestion
The final extension writes a little `Button is clicked.` text to the browser console and an altert when
a big red button is clicked.
```

jtpio commented on main-menu/README.md:
```diff
@@ -1,4 +1,6 @@
-# Menus: Extending the main app
+# Menus
+
+> Add a Menu in to the main app.
```
```suggestion
> Add a Menu to the main app.
```

jtpio commented on settings/README.md:
```diff
@@ -230,60 +238,61 @@ To react at a setting change by the user, you should use the signal
+      `Something went wrong when setting flag.\n${reason}`
+    );
+  });
+}
```
Are the first and last lines of the snippet missing?

echarles replied:
Done. Thx

jtpio commented on state/src/index.ts:
```diff
@@ -9,7 +9,7 @@ import { IStateDB } from '@jupyterlab/statedb';
 import { ReadonlyJSONObject } from '@lumino/coreutils';
 
-const PLUGIN_ID = 'state';
+const PLUGIN_ID = '@jupyterlab-examples/state:sate-example';
```
```suggestion
const PLUGIN_ID = '@jupyterlab-examples/state:state-example';
```

jtpio commented:
Thanks @echarles!

I left a couple of comments, as well as typo suggestions from previous iterations (that we take the opportunity to fix here too).

echarles commented:
Thx a lot @jtpio. I have committed your proposals and fixed the 2 remaining points.

jtpio commented:
Thanks.

Re-running CI now, as it looks like we are getting some random timeout failures with the browser check.

echarles commented:
yeah I have seen that. Timeouts look random. And rerunning from the UI fails all... I will push a small change now to retrigger.

jtpio commented:
Damn, looks like the same tests are failing.

jtpio commented:
Rerunning from the UI seems to trigger this "Checkout" issue with all of them failing...

echarles commented:
@jtpio Finally the failure were not random but due to the window.alert box that I have added. Chromium was waiting for a user interaction, hence the timeout. This is actually a good think as wee can keep our confidence in CI. I have removed the alerts on launch, which makes the CI green.

@fcollonval @jtpio Feel free to further review. I will create new preview images base which will align with the new namings.

jtpio commented:
> Finally the failures were not random but due to the `window.alert` box that I have added. Chromium was waiting for a user interaction, hence the timeout. This is actually a good think as wee can keep our confidence in CI. I have removed the alerts on launch, which makes the CI green.

Wow good find. It's really good that CI was able to catch this, the browser check is definitely useful :+1:

jtpio commented:
> Feel free to further review. I will create new preview images base which will align with the new namings.

Would you like to do this in the same PR or separately?

echarles commented:
I am running now browser check locally. It seems like you need to install a release and that the browser check does not work if you install dev (pip install -e .). With dev installed, browser check works for jlab sourcs, but not for 3rd party sources (at least for me).

I will push images tomorrow, so if you want to work on master before, just merge, I will open a new PR.

jtpio commented:
Thanks @echarles!

BTW I squashed the commits this time to remove the noise from the GitHub suggestions.

Otherwise it's fine to keep the default merge commit strategy for other cases.
